sdks/java/core/src/main/java/org/apache/beam/sdk/io/datastore/V1Beta3.java

```diff
@@ -166,104 +155,215 @@
  * @see org.apache.beam.sdk.runners.PipelineRunner
  */
 @Experimental(Experimental.Kind.SOURCE_SINK)
-public class DatastoreIO {
+public class V1Beta3 {
   /**
```

Comment:
Make constructor `protected` so that V1Beta3 itself cannot be directly instantiated by users (but still can be by `DatastoreIO`).